src/main/java/train/common/api/Locomotive.java

```diff
@@ -542,7 +585,7 @@ private double convertSpeed(double speed) {
 
 	public void soundHorn() {
 		for (EnumSounds sounds : EnumSounds.values()) {
-			if (sounds.getEntityClass() != null && !sounds.getHornString().equals("")&& sounds.getEntityClass().equals(this.getClass()) && whistleDelay == 0) {
+			if (sounds.getEntityClass() != null && sounds.getEntityClass().equals(this.getClass()) && whistleDelay == 0) {
```

Comment:
this part probably shouldn't have been removed, because the string can be empty, and if it is that makes a problem.